### PR TITLE
Added parsing for binary operators.

### PR DESCRIPTION
**Pull request by marcushanses**

### CODE REVIEW

File: parse/parse.ts [hunk 1]
```diff
@@ -40,3 +42,45 @@ export function add(
...
+export function parseExpression(source: string, handler?: Error.Handler): BinaryOperator
+export function parseExpression(source: string | Source, handler?: Error.Handler): BinaryOperator {
+	if (typeof source == "string") {
+		handler = handler || new Error.ConsoleHandler()
+		const tokens = lexer.tokenize(source, handler).toArray()
+		source = new Source(tokens, handler)
+	}
+	handler = handler instanceof Rule ? handler : undefined
+	const result = parseNextExpression(0, source)
+	return result
```
Comment (nilssonemma) [inline parse/parse.ts:56]: return parseNextExpression(0, source)

File: parse/Multiplication.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+		fetched = [
+			{
+				value: fetched.map(t => t.value).join(""),
+			},
+		]
+	return (
+		fetched &&
+		previous &&
+		(previous.precedence >= Multiplication.precedence
+			? parseNextExpression(new Multiplication(previous, new Value(fetched[0].value)), source)
```
Comment (nilssonemma) [inline parse/Multiplication.ts:17]: Returns the same thing in both cases

File: parse/parse.ts [hunk 3]
```diff
@@ -40,3 +42,45 @@ export function add(
...
+	) {
+		for (const expressionParser of expressionParsers) {
+			const r = expressionParser[0](source, left)
+			if (r) {
+				result = parseNextExpression(r, source)
+				break
+			}
+		}
+	}
+	return result || left || new Value(+(source.fetch()?.value ?? 3.14))
```
Comment (nilssonemma) [inline parse/parse.ts:75]: Remove fallback to 3.14 and handle undefined values differently

**Review by nilssonemma — CHANGES_REQUESTED** [review]

**Review by nilssonemma — APPROVED** [review]

File: parse/Subtraction.spec.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+describe("parse.Subtraction", () => {
+	it("9test rules related to addition", () => {
+		expect(selectively.parseExpression("5 - 6 * 7 - 5").toString()).toEqual("5 - 6 * 7 - 5")
+		expect(selectively.parseExpression("5 - 6 * 7 - 5 * 8").toString()).toEqual("5 - 6 * 7 - 5 * 8")
+		expect(selectively.parseExpression("5 * 5 - 5").evaluate()).toEqual(20)
+		expect(selectively.parseExpression("a * 5 - 5").evaluate({ a: 5 })).toEqual(20)
+	})
+	it("testet", () => {
+		expect(selectively.parseExpression("authorization.amount * 1.05 - authorization.captured.amount"))
```
Comment (fredrikbryntesson) [inline parse/Subtraction.spec.ts:11]: testet?

**Review by fredrikbryntesson — CHANGES_REQUESTED** [review]

**Review by fredrikbryntesson — APPROVED** [review]

**Pull request merged**